distributed/spans.py

```diff
@@ -68,15 +70,137 @@ class Span:
     #: ``distributed.extensions["spans"].spans[self.id[:-1]]``
     children: set[Span]
 
+    #: Task groups *directly* belonging to this span.
+    #:
+    #: See also
+    #: --------
+    #  traverse_groups
+    #:
+    #: Notes
+    #: -----
+    #: TaskGroups are forgotten when the last task is forgotten. If a user calls
+    #: compute() twice on the same collection, you'll have more than one group with the
+    #: same tg.key in this set! For the same reason, while the same TaskGroup object is
```

Comment:
nit: I think you mean `TaskGroup.name`?

```suggestion
    #: same tg.name in this set! For the same reason, while the same TaskGroup object is
```

Reply:
Out of scope: This sounds like it could generally cause all sorts of problems for us and third-party extensions tracking task groups, should we attach a unique key to them?

Reply:
The unique id already exists; it's `id()`